### ISSUE TITLE
Towards registering NEOs.jl

### ISSUE THREAD

**Issue by PerezHz** [PerezHz/NEOs.jl#3]
While trying to register NEOs.jl in the Julia General registry (see relevant NEOs.jl [commit](https://github.com/PerezHz/NEOs.jl/commit/d64a74741983bdb0692dc14ad3e1a4166e711a89#commitcomment-88570260), corresponding PR JuliaRegistries/General/pull/71434 and [job log](https://github.com/JuliaRegistries/General/actions/runs/3372030329/jobs/5594980311)) auto-merge didn't work due to several issues:

- [x] Name of package is not at least 5 characters long
- [x] The following dependencies do not have a [compat] entry that is upper-bounded and only includes a finite number of breaking releases: DataFrames, HORIZONS, IntervalArithmetic, PlanetaryEphemeris, Query, ReferenceFrameRotations, StaticArrays, StatsBase
- [x] I was not able to install the package (i.e. Pkg.add("NEOs") failed). See the CI logs for details.
- [x] I was not able to load the package (i.e. import NEOs failed). See the CI logs for details.
- [ ] Package name similar to 2 existing packages.
  - Similar to NES. Damerau-Levenshtein distance 2 is at or below cutoff of 2. Damerau-Levenshtein distance 1 between lowercased names is at or below cutoff of 1.
  - Similar to NExOS. Damerau-Levenshtein distance 2 is at or below cutoff of 2. Damerau-Levenshtein distance 1 between lowercased names is at or below cutoff of 1.

I'm opening this issue to try to resolve those issues, so that we can finally register NEOs.jl.

Besides discussing the name of the package (I would be in favor of keeping NEOs, but maybe NearEarthAsteroids could be an alternative), I think the other important points would be to fix the compat entries and using consistent bounds on TaylorSeries.jl and TaylorIntegration.jl throughout so that `Pkg.add` and `Pkg.import` work.

cc: @lbenet @LuEdRaMo

**Comment by PerezHz** [comment]
Actually, now that I think about it, since TaylorIntegration has a bound on TaylorSeries while re-exporting it, maybe we can get rid of the explicit dependency on TaylorSeries both here and in PlanetaryEphemeris.jl?

**Comment by lbenet** [comment]
Thanks for opening this issue!

TLDR: upgrading TaylorSeries is ok; let's not do it yet for TaylorIntegration.

Regarding the compatibility with TaylorSeries.jl, as far as @LuEdRaMo has looked, there is no problem of compatibility; see [this comment](https://github.com/PerezHz/NEOs.jl/pull/2#issuecomment-1288985634). I think to include versions v0.11 and v0.12 of TaylorSeries should then be ok.

TaylorIntegration.jl is more subtle in two respects: `@taylorize` by itself (with the current version of TI used in NEOs.jl) is not producing proper code to be used in the integration. This is related to one internal method which imposes using the same `Tayor1{T}` for the operands, while the operands are `Taylor1{Float64}` and `Taylor1{TaylorN{Float64}`. Credit for this investigation goes to @LuEdRaMo. We are checking now, if by converting the `Taylor1{Float64}` to the proper `Taylor1{TaylorN{Float64}` solves this problem. (The `Taylor1{Float64}` come from the integration of PlanetaryEphemeris.)

The second issue is a bit more subtle, and it is related to the changes within `@taylorize` (in the newer version of TI) which seems to introduce some regression (!?). At the moment we believe it is related to the temporal arrays, in particular those that represent arrays of `Taylor1`s, but may also be something different. While we are working on this it is not quite clear if this is truly problem...

Putting this together, I would suggest not to upgrade `TaylorIntegration` for the moment.

**Comment by lbenet** [comment]
Regarding the package name, I also like NEOs.jl, but the alternative is ok too.
